### PR TITLE
"Add chat functionality to projects page"

### PR DESCRIPTION
**Pull request by seheon99**
This pull request adds chat functionality to the projects page. It includes the following changes:

- Added ChatToolbox component for sending messages in chat rooms

- Added ChatUser type

- Added useChatSubscription hook for subscribing to chat messages

- Created ChatSocket class for handling chat socket connections

- Added ChatMessage, TextMessage, and ImageMessage interfaces for defining chat message types

- Updated ProfileCard component to display profile images from a cloud storage URL

- Created ChatStore for managing chat state

- Added chat store to the root store

- Updated ProjectsPage component to include chat functionality

- Added chat store to ProjectsPage component

- Updated gitignore to exclude chat.ts and chat.js files

- Added chat.ts file to stores directory

- Added empty-fetch-state.ts file to entities directory

- Added fetch-state.ts file to types directory

- Added chat.ts file to types directory

- Added useChat hook to hooks directory

- Updated package.json to include socket.io-client dependency

### CODE REVIEW

File: src/components/molecules/ChatBubble.tsx [hunk 1]
```diff
@@ -1,23 +1,53 @@
...
+        <Txt size="typo5" weight="bold" color="#757575">
+          {user.nickname}
+        </Txt>
+        <TextBubble myBubble={myBubble} size="typo4" weight="regular" color="#424242" {...props}>
+          {text}
+        </TextBubble>
+      </BubbleContainer>
+      <BubbleTime size="typo6" weight="medium" color="#9E9E9E">
+        12:24 PM
+      </BubbleTime>
```
Comment (jaewoongs) [inline src/components/molecules/ChatBubble.tsx:27]: P4 : 이거는 아직 백엔드에서 시간을 보내주지 않는건가요??

**Review by jaewoongs — DISMISSED** [review]

File: src/app/login-guard.tsx [hunk 2]
```diff
@@ -87,7 +87,7 @@ export const LoginGuard: React.FC<LoginGuardProps> = ({ children }) => {
   return (
     <>
       {children}
-      {isPopupOpened && popupComponent && <Backdrop>{popupComponent}</Backdrop>}
+      {/* {isPopupOpened && popupComponent && <Backdrop>{popupComponent}</Backdrop>} */}
     </>
```
[the remark this answers is not in the document]
Comment (seheon99) [reply src/app/login-guard.tsx:91]: 🫢

File: src/components/molecules/ChatBubble.tsx (hunk 1 above)
Comment (seheon99) [reply src/components/molecules/ChatBubble.tsx:27]: 지금 응답에는 빠져있어요!

Comment (jaewoongs) [reply src/components/molecules/ChatBubble.tsx:27]: 그럼 우선 머지하고 나중에 수정되면 작업을 하죠~

Comment (seheon99) [reply src/components/molecules/ChatBubble.tsx:27]: 조아용~

Comment (seheon99) [reply src/components/molecules/ChatBubble.tsx:27]: #100

**Review by jaewoongs — APPROVED** [review]

**Pull request merged**